macro-pdfviewer-api/src/main/java/com/xwiki/pdfviewer/filter/MJSFilter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.pdfviewer.filter;
+
+import java.lang.reflect.Field;
+
+import javax.inject.Singleton;
+
+import org.apache.tika.Tika;
+import org.apache.tika.config.TikaConfig;
+import org.apache.tika.mime.MediaType;
+import org.apache.tika.mime.MimeTypes;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.tika.internal.TikaUtils;
+
+/**
+ * Filter component for checking the mime type of .mjs files.
```

Comment:
It's not clear to me why you call it a "filter". What does this component filter? If the goal of this component is to register the ``msj`` mime type then it may be clearer to call it ``MJSMimeTypeRegistrator`` or something like that. Also, the JavaDoc should be updated, because the goal is not to check the mime type but to register it, if it's not already registered.